### PR TITLE
ref(calltree): fix scroll performance

### PR DESCRIPTION
**Pull request by JonasBa**
Seems like event handlers cause choppy scroll experiences, this PR employs the same optimization that react-virtualized and other popular libs are using which is to disable event handlers via (pointer-events: none) while a user is scrolling. Uses raf to try and schedule the micro task just before we are rendering elements to the dom

https://user-images.githubusercontent.com/9317857/171879058-8ba2638f-d8c6-4750-af5b-7c0b4722bb3b.mp4

### CODE REVIEW

File: static/app/utils/profiling/hooks/useVirtualizedTree/useVirtualizedTree.tsx [hunk 1]
```diff
@@ -5,6 +5,45 @@ import {useEffectAfterFirstRender} from 'sentry/utils/useEffectAfterFirstRender'
...
+  delay: number
+): AnimationTimeoutId => {
+  let start;
+  // wait for end of processing current event handler, because event handler may be long
+  Promise.resolve().then(() => {
+    start = Date.now();
+  });
+
+  const timeout = () => {
+    if (Date.now() - start >= delay) {
```
Comment (Zylphrex) [inline static/app/utils/profiling/hooks/useVirtualizedTree/useVirtualizedTree.tsx:33]: could `start` be `undefined` here if this handler is called before the `.then` is called?

File: static/app/utils/profiling/hooks/useVirtualizedTree/useVirtualizedTree.tsx [hunk 2]
```diff
@@ -291,14 +320,74 @@ export function useVirtualizedTree<T extends TreeLike>(
...
+  for (const item of items) {
+    renderedItems.push(
+      renderRow(item, {
+        handleRowClick: handleRowClick(item.key),
+        handleExpandTreeNode,
+        handleRowKeyDown,
+        tabIndexKey,
+      })
+    );
+  }
```
Comment (Zylphrex) [inline static/app/utils/profiling/hooks/useVirtualizedTree/useVirtualizedTree.tsx:383]: How many items can there be? Should this be memoized?

File: static/app/utils/profiling/hooks/useVirtualizedTree/useVirtualizedTree.tsx [hunk 3]
```diff
@@ -291,14 +323,87 @@ export function useVirtualizedTree<T extends TreeLike>(
...
+  // Basic styles for the element container. We fake the height so that the
+  // scrollbar is sized according to the number of items in the list.
+  const containerStyles: React.CSSProperties = useMemo(() => {
+    const height = tree.flattened.length * props.rowHeight;
+    return {height, maxHeight: height, overflow: 'hidden'};
+  }, [tree.flattened.length, props.rowHeight]);
+
+  // It is important that this is not executed from a map function because
+  // we are assigning the refs to each individual item. If we do that,
+  // we lose access to the refs and cannot call focus or scrollIntoView on them
```
Comment (Zylphrex) [inline static/app/utils/profiling/hooks/useVirtualizedTree/useVirtualizedTree.tsx:375]: Is this comment still true?

**Review by Zylphrex — APPROVED** [review]

File: static/app/utils/profiling/hooks/useVirtualizedTree/useVirtualizedTree.tsx (hunk 3 above)
Comment (JonasBa) [reply static/app/utils/profiling/hooks/useVirtualizedTree/useVirtualizedTree.tsx:375]: Yes, but it's in a bad place and is worded badly

**Pull request merged**